### PR TITLE
[Quest API] Add SendToInstance(instance_type, zone_short_name, instance_version, x, y, z, heading, instance_identifier, duration) to Perl/Lua.

### PR DESCRIPTION
**Pull request by KinglyKrab**
- Converts [this](https://github.com/ProjectEQ/projecteqquests/blob/master/plugins/Instances.pl#L4) script to a C++ method and adds Raid functionality.
- Add $client->SendToInstance(instance_type, zone_short_name, instance_version, x, y, z, heading, instance_identifier, duration) to Perl.
- Add client:SendToInstance(instance_type, zone_short_name, instance_version, x, y, z, heading, instance_identifier, duration) to Lua.

### CODE REVIEW

**Review by splose — APPROVED** [review]

File: zone/client.cpp [hunk 1]
```diff
@@ -10129,3 +10129,50 @@ void Client::SetAFK(uint8 afk_flag) {
...
+		uint32 raid_id = (GetRaid() ? GetRaid()->GetID() : 0);
+		instance_type_name = itoa(raid_id);
+	} else if (current_instance_type.find("guild") != std::string::npos) {
+		uint32 guild_id = (GuildID() > 0 ? GuildID() : 0);
+		instance_type_name = itoa(guild_id);
+	}
+
+	std::string full_bucket_name = fmt::format(
+		"{}_{}_{}",
+		instance_type_name,
```
Comment (Akkadius) [inline zone/client.cpp:10154]: `current_instance_type` should be included in the naming of this key not only for debugging / identifying purposes but it is possible for each of these different types of ID's to collide with each other

File: zone/client.cpp [hunk 2]
```diff
@@ -10129,3 +10129,50 @@ void Client::SetAFK(uint8 afk_flag) {
 	entity_list.QueueClients(this, outapp);
 	safe_delete(outapp);
 }
+
+void Client::SendToInstance(std::string instance_type, std::string zone_short_name, uint32 instance_version, float x, float y, float z, float heading, std::string instance_identifier, uint32 duration) {
+	uint32 account_id = AccountID();
```
Comment (Akkadius) [inline zone/client.cpp:10134]: `account_id` is unused (copy pasta)

File: zone/client.cpp [hunk 3]
```diff
@@ -10129,3 +10129,50 @@ void Client::SetAFK(uint8 afk_flag) {
...
+	std::string account_name = AccountName();
```
Comment (Akkadius) [inline zone/client.cpp:10135]: `account_name` is unused (copy pasta)

**Review by Akkadius — COMMENTED** [review]
Left a few comments, I'm excited for this to replace the existing plugin!

I'm sure its on your list already to have the existing plugin just call this native function

**Comment by KinglyKrab** [comment]
Will take a look this afternoon.

**Comment by KinglyKrab** [comment]
Pushed adjustments and resolved conflicts.

**Review by Akkadius — APPROVED** [review]

**Pull request merged**